flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -150,7 +150,35 @@ class Table(
     * }}}
     */
   def filter(predicate: Expression): Table = {
-    new Table(tableEnv, Filter(predicate, logicalPlan).validate(tableEnv))
+
+    predicate match {
```

Comment:
As far as I know `IN` is a general expression, so it can be used where ever expressions are allowed. I quickly checked it in MySQL and `SELECT field1 IN ( 22, 23 ) FROM  Table1` is not a problem.